src/Application/Implementation/StatePool.cs

```diff
@@ -30,9 +30,17 @@ public void Add(IExecutable executable)
             {
                 try
                 {
-                    foreach (var child in executable.Run())
-                        if (_exception == null)
-                            Add(child);
+                    foreach (var forks in executable.Run())
+                    {
+                        if (_exception != null)
```

Comment:
Does this bring back the behaviour we had before #85 where an exception on any 'leg' would signal an issue and all of the other 'legs' would discover this immediately on their next instruction call. That would be good. Does this implementation do that or do we need something slightly different in the looping and breaking here?

Reply:
Yes it does bring back that behaviour. I think that's one of the pluses for this approach, although imo it's a minor thing as there are other ways to terminate a loop anyway and it's not like this behaviour is *that* important in all scenarios.